src/meta/src/manager/sink_coordination/coordinator_worker.rs

```diff
@@ -523,24 +535,168 @@ impl CoordinatorWorker {
+                                    if next_committed_epoch >= epoch {
+                                        break;
+                                    }
+                                } else {
+                                    return Err(anyhow!(
+                                        "Hummock committed epoch sender closed unexpectedly"
+                                    ));
+                                }
+                            }
+                        }
+
+                        let mut delay = Duration::from_secs(1);
+                        let max_backoff = Duration::from_secs(60);
+                        let mut attempt = 0;
+
+                        loop {
+                            let start_time = Instant::now();
+                            attempt += 1;
+                            let commit_res = run_future_with_periodic_fn(
+                                coordinator.commit(epoch, commit_metadata.clone()),
+                                Duration::from_secs(5),
+                                || {
+                                    warn!(
+                                        elapsed = ?start_time.elapsed(),
+                                        %sink_id,
+                                        "committing"
+                                    );
+                                },
+                            )
+                            .await;
+
+                            match commit_res {
+                                Ok(_) => break,
+                                Err(e) => {
+                                    tracing::error!(
+                                        error = %e.as_report(),
+                                        %sink_id,
+                                        "failed to commit epoch {}, attempt {}. Retrying after {:?}",
+                                        epoch,
+                                        attempt,
+                                        delay
+                                    );
+                                    sleep(delay).await;
+                                    delay = std::cmp::min(delay * 2, max_backoff);
+                                }
+                            }
+                        }
+
+                        mark_record_committed(&db, sink_id as _, epoch).await?;
+                    }
+                }
                 prev_commit_epoch = Some(epoch);
             }
         }
     }
+
+    /// Return the log store rewind start epoch if exists.
+    async fn init_state_from_store(
+        &mut self,
+        db: &DatabaseConnection,
+        sink_id: SinkId,
+        coordinator: &mut SinkCommitCoordinator,
+    ) -> anyhow::Result<Option<u64>> {
+        match coordinator {
+            SinkCommitCoordinator::SinglePhase(coordinator) => {
+                coordinator.init().await?;
+                Ok(None)
+            }
+            SinkCommitCoordinator::TwoPhase(coordinator) => {
```

Comment:
The propagation of the first barrier of a newly create job and its checkpoint is expected not to be blocked. Therefore, when we see that the epoch list is empty, maybe we can get the first committed epoch from the CommittedEpochSubscriber, and then initialize a committed row in meta store table, and then in later code we can always treat like we have an initialized committed epoch.